amxmodx/commandline.cpp

```diff
@@ -0,0 +1,100 @@
+#include "amxmodx.h"
+
+#define CHECK_COMMANDLINE_AVAILABILITY()                                        \
+    if (!IsCommandLineAvailable())                                              \
+    {                                                                           \
+        LogError(amx, AMX_ERR_NATIVE, "The command line is not available.");    \
+        return 0;                                                               \
+    }
+
+static bool IsCommandLineAvailable()
+{
+    return g_engfuncs.pfnCheckParm != nullptr;
+}
+
+// native bool:IsCommandLineAvailable();
+static cell AMX_NATIVE_CALL IsCommandLineAvailable(AMX *amx, cell *params)
+{
+    return static_cast<cell>(IsCommandLineAvailable());
+}
+
+// native bool:FindCommandLineParam(const param[]);
+static cell AMX_NATIVE_CALL FindCommandLineParam(AMX *amx, cell *params)
+{
+    enum {arg_count, arg_param};
+
+    CHECK_COMMANDLINE_AVAILABILITY();
+
+    int length;
+    const char *param = get_amxstring(amx, params[arg_param], 0, length);
+    if (length == 0)
```

Comment:
What is the purpose of `length` check?

Reply:
In case the param is empty (for some reasons) to avoid unnecessary code execution in the engine, even though it doesn't matter, but since you got the length already available, there is no reason to no use it.

Reply:
@Arkshine it is a bit confusing, it makes me think like `EngCheckParm` can produce errors if parameter is empty string. So it should be explicitly commented with something like this: `// For optimization purposes`.

Reply:
Do we really have to do this?

Reply:
I can't understand why we need this optimization in function that is not critical to performance. And also who will call this function with empty param?

Reply:
I can't understand why it's so dramatic to add such check. The engine code doesn't check for an empty string and will loop/compare for nothing. We have an unused length variable available which can be used, even if it's not critical and probability someone passes an empty string is low, why it's so bad to use it?

Reply:
@Arkshine I need just a comment that this check doesn't change function logic.